### PR TITLE
Create database models

### PR DESCRIPTION
**Pull request by simplegoose**
In this PR the following was done:

- Models user, post, comment and like were created.
- Migrations to rename column posts_id were created to rename the column to post_id.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (Munsa1) [inline app/models/user.rb]: - [x] Kindly explicitly specify the foreign key for each association. This allows you to use custom column names for the foreign keys, 
Check the following example

```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```
Comment (Munsa1) [inline app/models/post.rb]: - [x] Kindly create a private method that updates the post as per requirement.

e.g

```
private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Comment (Munsa1) [inline app/models/post.rb]: - [x] Kindly add a rails active call back `after_save` to call the `#update_post_counter` method every time a new post has been saved as per requirement. Also, do this for the `update_likes_count` method as well as the `update_comments_count` method in the like model and comment model respectively

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @simplegoose , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Clean Codebase :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :users_id
+  belongs_to :post, class_name: 'Post'
+  after_save :update_likes_count
+
+  def update_likes_count
+    post.increment!(:likes_counter)
+  end
```
Comment (Qoosim) [inline app/models/like.rb:8]: - [OPTIONAL] Great job implementing the required changes from the previous reviewer. Kindly make this method to be `private`. It is considered good practice to declare callback methods as private. If left public, they can be called from outside of the model and violate the principle of object encapsulation.

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :users_id
+  belongs_to :post, class_name: 'Post'
+  after_save :update_comments_count
+
+  def update_comments_count
+    post.increment!(:comment__counter)
+  end
```
Comment (Qoosim) [inline app/models/comment.rb:8]: - [OPTIONAL] Great job implementing the required changes from the previous reviewer. Kindly make this method to be `private`. It is considered good practice to declare callback methods as private. If left public, they can be called from outside of the model and violate the principle of object encapsulation.

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @simplegoose, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**